libevmasm/Assembly.cpp

```diff
@@ -971,6 +972,84 @@ void appendBigEndianUint16(bytes& _dest, ValueT _value)
 	assertThrow(_value <= 0xFFFF, AssemblyException, "");
 	appendBigEndian(_dest, 2, static_cast<size_t>(_value));
 }
+
+uint16_t calculateMaxStackHeight(AssemblyItems const& _items, uint16_t _args)
```

Comment:
Since this calculation only makes sense for a complete code section, and not just any random collection of stack items, why not make it accept a `CodeSection`?

```suggestion
uint16_t calculateMaxStackHeight(CodeSection const& _section)
```

It won't change how it works but will add some context necessary for understanding why. It would also be helpful to link to the EIP that describes the stack validation algorithm.

Reply:
> It would also be helpful to link to the EIP that describes the stack validation algorithm.

Can you add a comment with the link to the EIP as well?

Reply:
done